vllm/entrypoints/openai/serving_engine.py

```diff
@@ -397,7 +397,52 @@ def _tokenize_prompt_input_or_inputs(
                     truncate_prompt_tokens=truncate_prompt_tokens,
                 )
 
-    def _preprocess_completion(
+    async def _tokenize_prompt_input_async(
+        self,
+        request: AnyRequest,
+        tokenizer: AnyTokenizer,
+        prompt_input: Union[str, List[int]],
+        truncate_prompt_tokens: Optional[Annotated[int, Field(ge=1)]] = None,
+        add_special_tokens: bool = True,
+    ) -> TextTokensPrompt:
+        return await make_async(self._tokenize_prompt_input)(
+            request=request,
+            tokenizer=tokenizer,
+            prompt_input=prompt_input,
+            truncate_prompt_tokens=truncate_prompt_tokens,
+            add_special_tokens=add_special_tokens)
+
+    async def _tokenize_prompt_inputs_async(
+        self,
+        request: AnyRequest,
+        tokenizer: AnyTokenizer,
+        prompt_inputs: Iterable[Union[str, List[int]]],
+        truncate_prompt_tokens: Optional[Annotated[int, Field(ge=1)]] = None,
+        add_special_tokens: bool = True,
+    ) -> Iterator[TextTokensPrompt]:
+        return await make_async(self._tokenize_prompt_inputs)(
+            request=request,
+            tokenizer=tokenizer,
+            prompt_inputs=prompt_inputs,
+            truncate_prompt_tokens=truncate_prompt_tokens,
+            add_special_tokens=add_special_tokens)
+
+    async def _tokenize_prompt_input_or_inputs_async(
+        self,
+        request: AnyRequest,
+        tokenizer: AnyTokenizer,
+        input_or_inputs: Union[str, List[str], List[int], List[List[int]]],
+        truncate_prompt_tokens: Optional[Annotated[int, Field(ge=1)]] = None,
+        add_special_tokens: bool = True,
+    ) -> Iterator[TextTokensPrompt]:
+        return await make_async(self._tokenize_prompt_input_or_inputs)(
+            request=request,
+            tokenizer=tokenizer,
+            input_or_inputs=input_or_inputs,
+            truncate_prompt_tokens=truncate_prompt_tokens,
+            add_special_tokens=add_special_tokens)
```

Comment:
I think it would be simpler if we just call `make_async` and assign these methods in `__init__`.

Reply:
got it. Like in [multiproc_gpu_executor.py](https://github.com/vllm-project/vllm/blob/ed46f143212203b7afcbc8538119b6e8155c643e/vllm/executor/multiproc_gpu_executor.py#L219) for example.